NumLib/ODESolver/NonlinearSolver-impl.h

```diff
@@ -149,6 +165,10 @@ solve(Vector &x)
         }
     }
 
+    MathLib::GlobalMatrixProvider<Matrix, Vector>::provider.releaseMatrix(J);
```

Comment:
These releasing functions appear more than once in different functions. Might be good to put them together in a single function.